src/natcap/invest/routedem.py

```diff
@@ -94,15 +92,168 @@
+                        "the stream segment."),
+                    "type": "number",
+                    "units": u.pixels,  # TODO: does this work?
+                },
+                "thresh_fa": {
+                    "about": (
+                        "The final threshold flow accumulation value used to "
+                        "determine the river segments."),
+                    "type": "number",
+                    "units": u.pixels,  # TODO: does this work?
+                },
+                "upstream_d8_dir": {
+                    "about": (
+                        "The direction of flow immediately upstream."),
+                    "type": "number",
+                    "units": u.none,
+                },
+                "ds_x": {
+                    "about": "The raster X coordinate for the outlet.",
+                    "type": "number",
+                    "units": u.none,
+                },
+                "ds_y": {
+                    "about": "The raster Y coordinate for the outlet.",
+                    "type": "number",
+                    "units": u.none,
+                },
+                "ds_x_1": {
+                    "about": (
+                        "The raster X coordinate that is 1 pixel upstream "
+                        "from the outlet."),
+                    "type": "number",
+                    "units": u.none,
+                },
+                "ds_y_1": {
+                    "about": (
+                        "The raster Y coordinate that is 1 pixel upstream "
+                        "from the outlet."),
+                    "type": "number",
+                    "units": u.none,
+                },
+                "us_x": {
+                    "about": (
+                        "The raster X coordinate for the upstream inlet."),
+                    "type": "number",
+                    "units": u.none,
+                },
+                "us_y": {
+                    "about": (
+                        "The raster Y coordinate for the upstream inlet."),
+                    "type": "number",
+                    "units": u.none,
+                },
+            },
+        },
+        "subwatersheds.gpkg": {
+            "about": (
+                "A GeoPackage with polygon features representing "
+                "subwatersheds.  A new subwatershed is created for each "
+                "tributary of a stream and is influenced greatly by "
+                "your choice of Threshold Flow Accumulation value."),
+            "geometries": spec_utils.POLYGON,
+            "fields": {
+                "stream_id": {
+                    "about": (
+                        "A unique stream id, matching the one in the Strahler "
+                        "stream order vector."),
+                    "type": "number",
+                    "units": u.none,
+                },
+                "terminated_early": {
```

Comment:
This is interesting. Have you encountered this? Is there a reason this could terminate early but we don't want to raise an `error`?

Reply:
This will happen if the boundary of the subwateshed exceeds 1 million, which is a really big subwatershed!  You can see this case in action at https://github.com/natcap/pygeoprocessing/blob/55fbbe9ef867308e26be71356b22e98059c2f897/src/pygeoprocessing/routing/routing.pyx#LL4209C37-L4209C37

This output like a detail of the neat algorithm Rich implemented here.